.github/workflows/main.yml

```diff
@@ -1,18 +1,24 @@
-name: Ruby
+name: Ruby Tests
 
-on: [push,pull_request]
+on: [push, pull_request]
 
 jobs:
-  build:
+  test:
+    name: Ruby ${{ matrix.ruby-version }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ['3.0.1', '3.1.1', '3.2.1', '3.3.1', '3.4.1']
```

Comment:
Went for this versions as the initially most stable versions after a major version release.
As it happens, the first `x.x.0` versions tend to be delivered at the end of the year and it often has some bugs that are tackled as part of the first minor version.